Fix bug in planning PruningVarExpand

fickludd opened the pull request:
Distinct is planned as an operator now and is recognized properly by the pruningVarExpand rewriter.

fickludd committed:
Fix bug in planning PruningVarExpand

Distinct is planned as an operator now and is recognized properly by the pruningVarExpand rewriter.

systay commented:
Nice find. Thanks for fixing my breakage... 🥇 

If I were to wish for one more thing, it would be to add at least one test to `PruningVarExpanderTest`

fickludd committed:
Update PruningVarExpanderTest to provide correct input plans.

fickludd commented:
@systay The blame should be on both of us for never adding these integration tests in the first place, but now it should be all good :).

Updated the `PruningVarExpanderTest` to actually test using relevant input plans, good call.

OliviaYtterbrink committed:
Expanded acceptance tests

OliviaYtterbrink committed:
Disabled slotted for LDBC Query 9

query 9 can be solved with a pruning var expand which is not supported
in slotted runtime.

sherfert commented on enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/VarLengthExpandQueryPlanAcceptanceTest.scala:
```diff
@@ -28,56 +28,101 @@ class VarLengthExpandQueryPlanAcceptanceTest extends ExecutionEngineFunSuite wit
+    result.executionPlanDescription() should useOperatorWithText("VarLengthExpand(All)", "(e)<-[:*..4]-(a)")
+    result.executionPlanDescription() should useOperatorWithText("NodeByLabelScan", ":To")
   }
 
```
Since we're porting everything to CypherComparisonSupport, does it make sense to write these new tests using that framework from the beginning?

systay replied:
I think we should move the tests over, but at the same time it would be nice to disconnect the two work streams
